### DISCUSSION TITLE
Channel output when restart time does not match model start time

### DISCUSSION THREAD

**Discussion by jmccreight** [NCAR/wrf_hydro_nwm_public#636] [Development]
There is a larger discussion of if/how/when/why to handle the situation when the namelist requests are restart file who's time does not match the specified model start time. Is this a bug or a feature?

However, there is a somewhat undesirable side-effect that should be handled: the RTOUT*DOMAIN (and CHANOBS) files are timestamped with the model time BUT the times in the files are based on the restart times: so they are actually consistent with the restart time. Meanwhile, the LSM is consistent with the model start time.

To me, this behavior implies that letting the model run when start time is not equal to restart time is a bug. I suppose this is debatable and everything is fine.... ?
## Example

Model start time:  9-12-13_00:00:00
Requested restart: 9-15-13_00:00:00
KHOUR=6
LSM timestep = 900 seconds
IOC configuration (channel_option=2, UDMP_OPT=1, among others).

```
jamesmcc@hydro-c1[995]:~/WRF_Hydro/FRNG_NHD/4DAY/NHDPLUS/run.pmoTruth.6Hour> ls *RTOUT*DOMAIN*
201309120015.CHRTOUT_DOMAIN2  201309120215.CHRTOUT_DOMAIN2  201309120415.CHRTOUT_DOMAIN2
201309120030.CHRTOUT_DOMAIN2  201309120230.CHRTOUT_DOMAIN2  201309120430.CHRTOUT_DOMAIN2
201309120045.CHRTOUT_DOMAIN2  201309120245.CHRTOUT_DOMAIN2  201309120445.CHRTOUT_DOMAIN2
201309120100.CHRTOUT_DOMAIN2  201309120300.CHRTOUT_DOMAIN2  201309120500.CHRTOUT_DOMAIN2
201309120115.CHRTOUT_DOMAIN2  201309120315.CHRTOUT_DOMAIN2  201309120515.CHRTOUT_DOMAIN2
201309120130.CHRTOUT_DOMAIN2  201309120330.CHRTOUT_DOMAIN2  201309120530.CHRTOUT_DOMAIN2
201309120145.CHRTOUT_DOMAIN2  201309120345.CHRTOUT_DOMAIN2  201309120545.CHRTOUT_DOMAIN2
201309120200.CHRTOUT_DOMAIN2  201309120400.CHRTOUT_DOMAIN2  201309120600.CHRTOUT_DOMAIN2
```

So the files have the model start time.

```
jamesmcc@hydro-c1[1005]:~/WRF_Hydro/FRNG_NHD/4DAY/NHDPLUS/run.pmoTruth.6Hour> ncdump -v time_observation 201309120015.CHRTOUT_DOMAIN2  | head -70 
...
// global attributes:
        :Conventions = "Unidata Observation Dataset v1.0" ;
        :cdm_datatype = "Station" ;
        :geospatial_lat_max = "90.0" ;
        :geospatial_lat_min = "-90.0" ;
        :geospatial_lon_max = "180.0" ;
        :geospatial_lon_min = "-180.0" ;
        :time_coverage_start = "2013-09-12_00:00:00" ;
        :stationDimension = "station" ;
        :missing_value = -9.e+15f ;
        :stream\ order\ output = 1 ;
        :time_coverage_end = "2013-09-12_00:15:00" ;
data:

 time_observation = 261000, 261000, 261000, 261000, 261000, 261000, 261000, 
...
```

All the time_observations are equal 261000. And yes, 

```
> as.POSIXct('2013-09-12 00:00:00')+261000
[1] "2013-09-15 00:30:00 MDT"
```

which corresponds to the difference of the model relative to the model RESTART time!
If all the hydro files are at this time... then OK, just fix the timestamps on the files. But the LSM is NOT at this time

```
jamesmcc@hydro-c1[1009]:~/WRF_Hydro/FRNG_NHD/4DAY/NHDPLUS/run.pmoTruth.6Hour> ncdump -v Times 201309120015.LDASOUT_DOMAIN3
...
 Times =
  "2013-09-12_00:15:00" ;
```

**Comment by gochis** [comment]
If I understand correctly, this is actually a feature.  In research there
are a number of times when we have used different spinup conditions for
different initialization times.  The idea behind this 'relaxation' was that
users should be able to use whatever time they want as a restart file and
it is incumbent upon them to make sure that the restart file specified in
the namelist is in fact the one that a user wants to use.  Savvy netcdf
users can always edit their netcdf data and attributes to work around a
tightly constrained system which insists on date matching between the
restart file and the initialization time but when Kevin from MMM
implemented this constraint in HRLDAS a few years ago, the backlash from
folks not doing opreational work with HRLDAS or WRF-Hydro suggested they
did not want that constraint/check and just wanted to 'point to any restart
file'.    Thus we relaxed the constraint.

Open for more discussion...

-dave

**Comment by jmccreight** [comment]
Sure. But then the hydro output times should be consistent with the run
start time like the LSM.

---

James L. McCreight
